pkg/lakom/config/config.go

```diff
@@ -87,9 +88,36 @@ func (c *Config) Complete() (*CompletedConfig, error) {
 		res.Keys = append(res.Keys, config)
 	}
 
+	res.Keys = uniqueKeys(res.Keys)
+
 	return &res, nil
 }
 
+// uniquekeys returns the unique set of [VerifierKey] items.
+func uniqueKeys(keys []VerifierKey) []VerifierKey {
+	result := make([]VerifierKey, 0)
+
+	for _, item := range keys {
+		// Equality check between the keys is achieved via the 'Equal'
+		// function that we know is implemented for every public key type.
+		// Ref: https://pkg.go.dev/crypto#PublicKey
+		// We cast to the interface to inform the compiler of the func.
+		key, ok := item.Key.(interface{ Equal(x crypto.PublicKey) bool })
+		if !ok {
+			continue
```

Comment:
What should be done in this case? Would a simple log be enough?

Reply:
> This should theoretically happen only if the specific PublicKey has no implementation of the `Equal` function. That's possible to happen but I'd assume is extremely unlikely since I'd expect that the PublicKey key types that the Cosign lib supports should all be compliant with this interface.

The issue is that this interface in question is not actually defined by the stdlib and is more like a good practice, that all PublicKeys _should_ implement it.

I would keep this check, just to make sure that the PublicKeys I work with do implement it.

Reply:
I think an error should be returned here. Or at least check in the `Complete` method that the keys do implement the interface. Another option is to change the type of `Verifier.Key` to the interface that `crypto.PublicKey` was supposed to be but it is not due to backward compatibility. In this case, you can be sure in `uniqueKeys` that the key implements the interface.

Reply:
> I think an error should be returned here.
 
In the case that a user supplies a public key type that has no implementation of `Equal`, and we throw an error, we'd effectively be disallowing them from using Lakom with the specific key that has the issue and they want to use. I'm worried that that might cause more issues in the future rather than make things better.

I personally think we should output some sort of warning. Though that would mean that duplicate keys would be possible in this case.

> Another option is to change the type of Verifier.Key to the interface that crypto.PublicKey was supposed to be but it is not due to backward compatibility. In this case, you can be sure in uniqueKeys that the key implements the interface.

Won't this just move the check earlier into the codepath? 
The thing I like with the current approach is that the topic of 'comparing keys' is contained completely inside this function. So the whole issue does not leak to any other types/functions/files and you don't need to know about these things unless you're trying to modify/use the `uniqueKeys` function.

Reply:
Until now it was not allowed end users to provide trusted keys, hence we can still implement restrictions what keys are supported without breaking any existing usage (there is still nobody using the feature :wink:)

The validation of the correctness of the provided keys should not be implemented only in the `lakom` component, it must be implemented also in the `shoot-lakom-admission` so that end user are not allowed to set key that does not implement the public key interface.